tests/src/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -103,10 +103,20 @@ public class CoreclrTestWrapperLib
 
         static bool CollectCrashDump(Process process, string path)
         {
-            using (var crashDump = File.OpenWrite(path))
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                var flags = DbgHelp.MiniDumpType.MiniDumpWithFullMemory | DbgHelp.MiniDumpType.MiniDumpIgnoreInaccessibleMemory;
-                return DbgHelp.MiniDumpWriteDump(process.Handle, process.Id, crashDump.SafeFileHandle, flags, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+                using (var crashDump = File.OpenWrite(path))
+                {
+                    var flags = DbgHelp.MiniDumpType.MiniDumpWithFullMemory | DbgHelp.MiniDumpType.MiniDumpIgnoreInaccessibleMemory;
+                    return DbgHelp.MiniDumpWriteDump(process.Handle, process.Id, crashDump.SafeFileHandle, flags, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+                }
+            }
+            else
+            {
+                ProcessStartInfo createdumpInfo = new ProcessStartInfo("createdump");
```

Comment:
Do you know if test programs hang directly off this process? Otherwise they might need sudo.

Reply:
For the dump-on-assert support on non-Windows, they invoke createdump from the process without using sudo.

It's possible that this might need sudo.

Reply:
They as in Arcade? With the env variable we don't need to because we set it up from a process hanging off directly from the hanging process.

Reply:
They as in coreclr.dll.